player/playloop.c

```diff
@@ -529,6 +529,9 @@ double get_playback_time(struct MPContext *mpctx)
         if (length >= 0)
             cur = MPCLAMP(cur, 0, length);
```

Comment:
oh, we already clamping in some cases

Reply:
Yeah this already existed for cosmetic purposes (seeking edgcase) so I figure it's the most logical place to add more special clamps.